packages/gateway/src/gateway-fetch-metadata.ts

```diff
@@ -57,6 +57,7 @@ const gatewayLanguageModelEntrySchema = z.object({
   description: z.string().nullish(),
   pricing: gatewayLanguageModelPricingSchema.nullish(),
   specification: gatewayLanguageModelSpecificationSchema,
+  modelType: z.enum(['language', 'embedding', 'image']).optional(),
```

Comment:
AI SDK prefers `nullish` -- may need edits to code where we reference, but best for us to follow esp. since we use nullish in other fields above